diff --git a/vendor/sqlite/src/sqlite3.c b/vendor/sqlite/src/sqlite3.c
index 08c593e55c..e3757f968b 100644
--- a/vendor/sqlite/src/sqlite3.c
+++ b/vendor/sqlite/src/sqlite3.c
@@ -39324,8 +39324,12 @@ static int unixLogErrorAtLine(
   ** available, the error message will often be an empty string. Not a
   ** huge problem. Incorrectly concluding that the GNU version is available
   ** could lead to a segfault though.
+  **
+  ** Forum post 3f13857fa4062301 reports that the Android SDK may use
+  ** int-type return, depending on its version.
   */
-#if defined(STRERROR_R_CHAR_P) || defined(__USE_GNU)
+#if (defined(STRERROR_R_CHAR_P) || defined(__USE_GNU)) \
+  && !defined(ANDROID) && !defined(__ANDROID__)
   zErr =
 # endif
   strerror_r(iErrno, aErr, sizeof(aErr)-1);
